### PR TITLE
Add flag to disable Vulkan support & support target-os for Emscripten.

### LINKED ISSUE #10424: [package] ffmpeg/5.0: Cannot build to WASM with Emscripten.
### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **ffmpeg/5.0**
  * Operating System+version: **Linux Manjaro cross to Web Assembly with Emscripten**
  * Compiler+version: **Clang 13**
  * Conan version: **conan 1.47.0**
  * Python version: **Python 3.10.4**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
arch=wasm
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=13
os=Emscripten

[options]
[build_requires]
*: emsdk/3.1.7

[env]
```

### Conanfile build options for ffmpeg
```
ffmpeg:with_zlib=False
ffmpeg:with_bzip2=False
ffmpeg:with_lzma=False
ffmpeg:with_libiconv=False
ffmpeg:with_freetype=False
ffmpeg:with_openjpeg=False
ffmpeg:with_openh264=False
ffmpeg:with_opus=False
ffmpeg:with_vorbis=False
ffmpeg:with_zeromq=False
ffmpeg:with_sdl=False
ffmpeg:with_libx264=False
ffmpeg:with_libx265=False
ffmpeg:with_libvpx=False
ffmpeg:with_libmp3lame=False
ffmpeg:with_libfdk_aac=False
ffmpeg:with_libwebp=False
ffmpeg:with_ssl=False
ffmpeg:with_programs=False
```

### Steps to reproduce (Include if Applicable)
`conan install .. -pr:h ../emscripten_conan_profile -pr:b default --build=ffmpeg` fails firstly because the `--target-os` value of `emscripten` is unrecognised by the ffmpeg build system and secondly because there isn't currently a flag to disable Vulkan support.

I have a branch which fixes both issues for me that I would like to open a PR for here:
https://github.com/simoncent/conan-center-index/tree/ffmpeg/emscripten-support

I'm *very* new to Conan, attempting to use Conan for my ffmpeg dependency was my first investigation into using it to replace all my manually rolled dependencies in Cmake, so I apologise if I've done something stunningly stupid and this issue/PR is unnecessary.

</details>

### PR DESCRIPTION
**Pull request by simoncent**
Specify library name and version:  **ffmpeg/5.0**

This PR fixes #10424.

Building with Emscripten fails firstly because the `--target-os` value of `emscripten` is unrecognised by the ffmpeg build system and secondly because there isn't currently a flag to disable Vulkan support.

I'm very new to Conan, attempting to use Conan for my ffmpeg dependency was my first investigation into using it to replace all my manually rolled dependencies in Cmake, so I apologise if I've done something stunningly stupid and this issue/PR is unnecessary.

I can build this PR locally for Emscripten with a profile similar to:
<details>
<summary>Click to show code</summary>

```
include(default)

[settings]
arch=wasm
build_type=Release
compiler=clang
compiler.libcxx=libc++
compiler.version=15
os=Emscripten

[options]
ffmpeg:with_zlib=False
ffmpeg:with_bzip2=False
ffmpeg:with_lzma=False
ffmpeg:with_libiconv=False
ffmpeg:with_freetype=False
ffmpeg:with_openjpeg=False
ffmpeg:with_openh264=False
ffmpeg:with_opus=False
ffmpeg:with_vorbis=False
ffmpeg:with_zeromq=False
ffmpeg:with_sdl=False
ffmpeg:with_libx264=False
ffmpeg:with_libx265=False
ffmpeg:with_libvpx=False
ffmpeg:with_libmp3lame=False
ffmpeg:with_libfdk_aac=False
ffmpeg:with_libwebp=False
ffmpeg:with_ssl=False
ffmpeg:with_programs=False

[build_requires]
*: emsdk/3.1.7

[env]
```

</details>

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by simoncent** [comment]
The build seems to have failed because of a connection issue, is there a way to kick off a new build without pushing a change?

**Comment by uilianries** [comment]
@simoncent I just restart it now. Have you signed the CLA? If not, please, it should be signed, otherwise, your PR will not be merged.

**Comment by simoncent** [comment]
Thanks for kicking it off. I'm just waiting for approval from my employer before I sign it.

**Comment by simoncent** [comment]
@uilianries Are these builds stuck or something? They've been running since last night!

**Comment by uilianries** [comment]
@simoncent I just canceled and restarted now. I'll follow its status. Thank you for reporting.

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
@simoncent Everything good now!

**Review by SSE4 — APPROVED** [review]

File: recipes/ffmpeg/all/conanfile.py
Comment (madebr) [inline recipes/ffmpeg/all/conanfile.py:391]: Would changing the code to let `self._target_os` return `none` when `os=Emscripten` make more sense?

Comment (SSE4) [reply recipes/ffmpeg/all/conanfile.py:391]: that might be too complicated - this part of the code is too sophisticated and fragile. I think it's fine.

**Pull request merged**
